consensus/clique/src/main/java/org/hyperledger/besu/consensus/clique/CliqueProtocolSchedule.java

```diff
@@ -56,9 +62,24 @@ public static ProtocolSchedule create(
 
     final EpochManager epochManager = new EpochManager(cliqueConfig.getEpochLength());
 
+    // TODO: roll EIP1559 object into fee market and remove these configs
+    final BaseFeeMarket londonFeeMarket =
+        FeeMarket.london(
+            targetGasLimit.stream()
```

Comment:
This doesn't feel right, or I'm reading it wrong.  This is taking what was a function without an input (how to calculate the gas limit) and replacing it with something that has a fixed input derived from a CLI flag.  

How do we update this value at runtime via RPC? 

Making operators restart their nodes to change the target is a loss of functionality.

Reply:
the `miner_changeTargetGasLimit` endpoint should continue to function as normal, with the exception being that it needs to update the targetGasLimit for all of the GasLimitCalculators, since there isn't the notion of 'change the targetGasLimit as of this block number'.  

Previously there was only one global gas limit calculator to update, and it was plumbed via BesuCommand.  Now however there can be multiple calculators, each of which applies to their respective protocolSpecs.

Reply:
The OptionalLong is an immutable value however, it is not AtomicLong which can be updated.  Once set there is no changing the GasLimitCalculator, the RPCs would need to replace it within the schedule.

What should change per schedule is the function to interpret the configured flag, not a one-time evaluation of the flag with per-schedule meaning.

Reply:
It is probably difficult to follow the change, but the actual targetGasLimit is non an OptionalLong, it is `private long`:

https://github.com/hyperledger/besu/pull/2632/files#diff-bf6252888b0185ced31a58f4ca5755847f8f913680d5fd23155bf364bc054d34R28

OptionalLong is just wrapping targetGasLimit because it is an optional parameter.  If there is not a targetGasLimit present, a constant gas limit calculator is used rather than a targeting gas limit calculator.

Reply:
I can change the mutable long to AtomicLong.  the targetGasLimit update within the protocol schedule is happening here:
https://github.com/hyperledger/besu/pull/2632/files#diff-a498e6c44084bb5267fd823cb8591538e6b7f99dbb21bf0fc55438ae5bb4c03bR38-R48

Reply:
Merging the CLI setting of the gas limit and the schedule interpretation of that limit are not two things I would expect to be in the same class.  

I think we would get better utility by changing org.hyperledger.besu.ethereum.blockcreation.GasLimitCalculator#nextGasLimit to take two args, the current block limit and the CLI configure limit (if configured), so `long` and `OptionalLong`.  That way instead of diving into the schedules (which IMHO should be 100% immutable once set) we set the value in the AbstractBlockCreator, or the AbstractBlockCreator has an object that provides that value (a shared `AtomicLong`, a `Provider<Long>`, a custom class, etc.  

It fels like the value should be in `MiningParametes` since changing the Coinbase has to follow similar plumbing?  Then the CLI configuraiton gets passed into the various MinerExecutors the same way Coinbase gets passed around.  Another option would be the various `MiningCoordinators`

Reply:
great call on MiningParameters.  That makes perfect sense to fold targetGasLimit into there, and keep gas limit calculators final/immutable